common/sagas/transaction/fields/fields.ts

```diff
@@ -47,5 +56,6 @@ export const fields = [
   takeEvery(TypeKeys.DATA_FIELD_INPUT, handleDataInput),
   takeEvery(TypeKeys.GAS_LIMIT_INPUT, handleGasLimitInput),
   takeEvery(TypeKeys.GAS_PRICE_INPUT, handleGasPriceInput),
-  takeEvery(TypeKeys.NONCE_INPUT, handleNonceInput)
+  takeEvery(TypeKeys.NONCE_INPUT, handleNonceInput),
+  takeLatest(TypeKeys.GAS_PRICE_INPUT_INTENT, handleGasPriceInputIntent)
```

Comment:
Instead of `takeLatest` + `call(delay ...` you should use `throttle`. Using `takeLatest` + `delay` wont actually reduce the amount of actions being dispatched, it'll just delay all of the actions by 300 ms. `throttle` implements a sliding buffer + delay with an action channel, the result being that every action that is dispatched while the throttled saga is paused is dropped, except for the latest one (which is exactly what you want)

Reply:
Totally my bad here, i was reading the code as `takeEvery` even though I mentioned `takeLatest`. This is fine